nixos/modules/services/hardware/throttled.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.throttled;
+in {
+  options = {
+    services.throttled = {
+      enable = mkEnableOption "fix for Intel CPU throttling";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.throttled ];
+
+    environment.etc."lenovo_fix.conf".source = "${pkgs.throttled}/etc/lenovo_fix.conf";
```

Comment:
Possibly https://github.com/NixOS/nixos-hardware is a better place for this

Reply:
I'm planning to teach `nixos-hardware` to enable it, but I think having the module and package in `nixpkgs` makes sense. I thought `nixos-hardware` was mostly supposed to just be configurations, with the software actually in `nixpkgs`.

Reply:
Yeah I meant to just add the package here, but add the module in nixos-hardware

Reply:
Not everyone with a ThinkPad uses `nixos-hardware`!

Reply:
I agree with @michaelpj. Since it is a package and module that lives in nixpkgs.

Reply:
I'm not sure if it's a good idea to have hardware-specific options in NixOS. Nixpkgs is too big and bloaty already, imagine having all of nixos-hardware available as options as well. nixos-hardware seems to have been created for exactly this reason, to not have to bloat nixpkgs with tons of hardware specific peculiarities.

Reply:
We have all sorts of obscure software in nixpkgs. I don't see why this is any different.

Re: `nixos-hardware`, I don't agree with your description of what it's for. The readme says "NixOS *profiles* covering hardware quirks." Now, we don't have a clear definition of what a "profile" is, but it seems that we usually use this to mean modules that just set other module options, and don't define any new functionality. That is, `nixos-hardware` is a convenient place to get a bunch of settings that work for your machine, but all the functionality comes from nixpkgs. You should be able to set it all up yourself without using `nixos-hardware`.

Maybe if/when we get flakes we'll want to revisit splitting up nixpkgs more, but I don't think doing it for random modules in the mean time is a good idea.

Reply:
> That is, nixos-hardware is a convenient place to get a bunch of settings that work for your machine, but all the functionality comes from nixpkgs. You should be able to set it all up yourself without using nixos-hardware

Once the package is in nixpkgs, that's easily done by adding this to your NixOS config:
```nix
{
  systemd.packages = [ pkgs.throttled ];
  environment.etc."lenovo_fix.conf".source = "${pkgs.throttled}/etc/lenovo_fix.conf";

  # Or if it were in nixos-hardware
  imports = [ <nixos-hardware/lenovo/throttle-fix.nix> ];
}
```

Part of my reason against putting such trivial and partly hardware-specific modules in nixpkgs is because of https://github.com/NixOS/nixpkgs/issues/57477, but I'm hoping for that to be solved eventually so I don't have to worry about that.

Anyways, my opinion is that hardware-specific things should live in nixos-hardware, but if people really want this in nixpkgs then so be it.

Reply:
This could make it hard to update `throttled` in nixpkgs without breaking the module in `nixos-hardware`. Nixos module options are an interface where we provide some sort of stability or compatibility anyway.